File: ECE-FIAD2 (Backend B)/Dominio/Entidades/Citas/Cita.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using Dominio.Entidades.Bases;
using Dominio.Entidades.Doctores;
using Dominio.Entidades.Pacientes;
using Dominio.Enumeraciones;
namespace Dominio.Entidades.Citas
{
    public class Cita : EntidadBase
    {
        public int IdPaciente { get; set; }
        public int IdDoctor { get; set; }
        public DateTime FechaHora { get; set; }
        public string Motivo { get; set; } = string.Empty;
        public EstadoCita Estado { get; set; } = EstadoCita.Pendiente;
        public string? Notas { get; set; }
        public virtual Paciente Paciente { get; set; } = null!;
        public virtual Doctor Doctor { get; set; } = null!;
        public void Confirmar()
        {
            Estado = EstadoCita.Confirmada;
            FechaDeModificacion = DateTime.UtcNow;
        }
        public void Cancelar()
        {
            Estado = EstadoCita.Cancelada;


            FechaDeModificacion = DateTime.UtcNow;
        }
        public void Completar()
        {
            Estado = EstadoCita.Completada;
            FechaDeModificacion = DateTime.UtcNow;
        }
        public void RegistrarNoAsistencia()
        {
            Estado = EstadoCita.NoAsistio;
            FechaDeModificacion = DateTime.UtcNow;
        }
    }
}

File: GestionClinica-Frontend-A/src/types/Cita/Cita.ts
import { type Doctor } from "../Doctor/Doctor";
import { type Paciente } from "../Paciente/Paciente";

export interface Cita {
    id: number;
    idPaciente: number;
    idDoctor: number;
    fechaHora: string;
    motivo: string;
    notas?: string;
    estado: number;
    paciente?: Paciente;
    doctor?: Doctor;
}

export interface CitaRespuestaB {
    id: number;
    idPaciente: number;
    nombrePaciente: string;
    idDoctor: number;
    nombreDoctor: string;
    nombreEspecialidad: string;
    fechaHora: string;
    motivo: string;
    notas?: string;
    estado: number;
    fechaCreacion: string;
    fechaModificacion?: string | null;
    fechaEliminacion?: string | null;
    eliminado: boolean;
}